spatialos/UnrealGDK
# UNR-1644: Schema compiler failure should fail schema generation

**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
UNR-1644: Change SpatialGDKGenerateSchema to cause the window to display "schema generation failed" if schema compilation failed.

#### Release note
Bugfix: When schema compiler fails, schema generation correctly shows an error.

#### Tests
Running schema generation normally succeeds as normal. Deleting GDK schema files and running an iterative schema gen correctly produces an output.

How can this be verified by QA? 

Run an iterative schema gen that should succeed and confirm the window displays the succeed message, then force the iterative schema gen to fail (e.g. by deleting GDK schema files) and confirming the window displays an error message.

Also, delete the schema descriptor directory in a test project (spatial/build/assembly/schema) and set its parent directory to read-only (right click on directory -> properties -> select security and deny write). Running the full schema gen should fail. Deleting schema descriptor directory without changing access permissions should still result in schema gen success.

#### Primary reviewers
@joshuahuburn @improbable-valentyn
